File: DOTNET_CSharp/Library/GridPaths.cs
using static Library.Math;

namespace Library {
    public static class GridExtensions {
        public static int CountUniquePathsConst(int rows, int columns) {
            --rows;
            --columns;
            return Factorial(rows + columns) 
                / (Factorial(rows) * Factorial(columns));
        }

        public static int CountUniquePathsDP(int rows, int columns) {
            var paths = new int[rows, columns];

            for (var i = 0; i < rows; i++) 
                paths[i, 0] = 1;
            
            for (var i = 0; i < columns; i++) 
                paths[0, i] = 1;
            
            for (var i = 1; i < rows; i++)
                for (var j = 1; j < columns; j++) 
                    paths[i, j] = paths[i - 1, j] + paths[i, j - 1];

            return paths[rows - 1, columns - 1];
        }
    }
}

File: FSharp/Library/GridPaths.fs
namespace Library

module GridPaths =
    open Math
    let CountUniquePathsConst rows columns =
        let r, c = (rows - 1, columns - 1)
        Factorial (r + c) / (Factorial(r) * Factorial(c))
    
    let CountUniquePathsDP rows columns =
        let paths = 
            Array2D.init rows columns (fun _ _ -> 0)
        
        for i = 0 to rows - 1 do 
            paths.[i, 0] <- 1
        
        for i = 0 to columns - 1 do 
            paths.[0, i] <- 1
        
        for i = 1 to rows - 1 do
            for j = 1 to columns - 1 do
                paths.[i, j] <- paths.[i - 1, j] + paths.[i, j - 1]
        
        paths.[rows - 1, columns - 1]